File: src/main/scala/org/lolhens/minechanics/client/config/gui/ConfigGuiFactory.scala
package org.lolhens.minechanics.client.config.gui

import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.fml.client.IModGuiFactory
import net.minecraftforge.fml.client.IModGuiFactory.{RuntimeOptionCategoryElement, RuntimeOptionGuiHandler}

class ConfigGuiFactory extends IModGuiFactory {
  override def getHandlerFor(arg0: RuntimeOptionCategoryElement): RuntimeOptionGuiHandler = null

  override def initialize(arg0: Minecraft) = {}

  override def mainConfigGuiClass(): Class[_ <: GuiScreen] = classOf[ConfigGui]

  override def runtimeGuiCategories(): java.util.Set[RuntimeOptionCategoryElement] = null
}

File: src/main/java/org/lolhens/minechanics/core/config/gui/ConfigGuiFactory.java
package org.lolhens.minechanics.core.config.gui;

import java.util.Set;

import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiScreen;
import cpw.mods.fml.client.IModGuiFactory;

public class ConfigGuiFactory implements IModGuiFactory {

    @Override
    public RuntimeOptionGuiHandler getHandlerFor(RuntimeOptionCategoryElement arg0) {
        return null;
    }

    @Override
    public void initialize(Minecraft arg0) {
    }

    @Override
    public Class<? extends GuiScreen> mainConfigGuiClass() {
        return ConfigGui.class;
    }

    @Override
    public Set<RuntimeOptionCategoryElement> runtimeGuiCategories() {
        return null;
    }

}
